### PR TITLE
Improvement course documentation and code comments for project 1

### LINKED ISSUE #113: refine project 1

### PR DESCRIPTION
**Pull request by rebelice**
close #113 

1. add code comments for project 1-2
2. add a `Makefile` commend for run test in project 1
3. refine course doc for project 1

### CODE REVIEW

**Comment by rebelice** [comment]
PTAL @pupillord

File: tablecodec/tablecodec.go [hunk 1]
```diff
@@ -71,7 +71,33 @@ func EncodeRowKeyWithHandle(tableID int64, handle int64) kv.Key {
 
 // DecodeRecordKey decodes the key and gets the tableID, handle.
 func DecodeRecordKey(key kv.Key) (tableID int64, handle int64, err error) {
-	/* Your code here */
+	/* Project 1-2: your code here
+	 * DecodeRecordKey decodes the key and gets the tableID, handle.
+	 * Decode is actually the reverse of encoding, so you can refer to EncodeRowKeyWithHandle.
+	 *
+	 * Parma
```
Comment (pupillord) [inline tablecodec/tablecodec.go]: There is a typo

File: tablecodec/tablecodec.go [hunk 2]
```diff
@@ -94,7 +120,34 @@ func EncodeIndexSeekKey(tableID int64, idxID int64, encodedValue []byte) kv.Key
 
 // DecodeIndexKeyPrefix decodes the key and gets the tableID, indexID, indexValues.
 func DecodeIndexKeyPrefix(key kv.Key) (tableID int64, indexID int64, indexValues []byte, err error) {
-	/* Your code here */
+	/* Project 1-2: your code here
+	 * DecodeIndexKeyPrefix decodes the key and gets the tableID, indexID, indexValues.
+	 * Decode is actually the reverse of encoding, so you can refer to EncodeIndexSeekKey.
+	 *
+	 * Parma
```
Comment (pupillord) [inline tablecodec/tablecodec.go]: The same typo

**Review by pupillord — COMMENTED** [review]
LGTM，but some typo.

**Comment by rebelice** [comment]
fixed typo and PTAL @pupillord

**Comment by pupillord** [comment]
LGTM

**Review by ngaut — DISMISSED** [review]

**Review by ngaut — APPROVED** [review]

**Pull request merged**